rollup-interface/src/state_machine/stf.rs

```diff
@@ -91,39 +106,41 @@ pub trait StateTransitionFunction<Vm: Zkvm, B: BlobTransactionTrait> {
     /// or validated together with proof during verification
     type Witness: Default + Serialize;
 
-    /// A proof that the sequencer has misbehaved. For example, this could be a merkle proof of a transaction
-    /// with an invalid signature
-    type MisbehaviorProof;
-
     /// Perform one-time initialization for the genesis block.
     fn init_chain(&mut self, params: Self::InitialState);
 
-    /// Called at the beginning of each **DA-layer block** - whether or not that block contains any
+    /// Called at each **DA-layer block** - whether or not that block contains any
     /// data relevant to the rollup.
     /// If slot is started in Full Node mode, default witness should be provided.
     /// If slot is started in Zero Knowledge mode, witness from execution should be provided.
-    fn begin_slot(&mut self, witness: Self::Witness);
-
-    /// Apply a blob/batch of transactions to the rollup, slashing the sequencer who proposed the blob on failure.
-    /// The concrete blob type is defined by the DA layer implementation, which is why we use a generic here instead
-    /// of an associated type.
+    ///
+    /// Applies batches of transactions to the rollup,
+    /// slashing the sequencer who proposed the blob on failure.
+    /// The concrete blob type is defined by the DA layer implementation,
+    /// which is why we use a generic here instead of an associated type.
+    ///
     /// Misbehavior hint allows prover optimizations - the sequencer can be slashed
     /// for including a transaction which fails stateless checks (i.e. has an invalid signature) -
     /// and in that case we ignore his entire batch.
     /// This method lets you give a hint to the prover telling
     /// it where that invalid signature is, so that it can skip signature checks on other transactions.
-    /// (If the misbehavior hint is wrong, then the host is malicious so we can
+    /// (If the misbehavior hint is wrong, then the host is malicious, so we can
     /// just panic - which means that no proof will be created).
-    fn apply_blob(
-        &mut self,
-        blob: &mut B,
-        misbehavior_hint: Option<Self::MisbehaviorProof>,
-    ) -> BatchReceipt<Self::BatchReceiptContents, Self::TxReceiptContents>;
-
-    /// Called once at the *end* of each DA layer block (i.e. after all rollup blobs have been processed)
-    /// Commits state changes to the database
     ///
-    fn end_slot(&mut self) -> (Self::StateRoot, Self::Witness);
+    /// Commits state changes to the database
+    fn apply_block<'a, I>(
+        &mut self,
+        height: u64,
```

Comment:
Let's remove this argument, since if the prover lies we don't have a way to detect it. We can still get access to the height inside the STF using the chain state module

Reply:
Yes, and I think it is not blocking this PR, it will only be needed for soft confirmations PR.